sdk/ai/azure-ai-projects/tests/test_base.py

```diff
@@ -0,0 +1,231 @@
+        "dataset_name_3": f"test-dataset-name-{random.randint(0, 99999):05d}",
+        "dataset_name_4": f"test-dataset-name-{random.randint(0, 99999):05d}",
+        "dataset_version": 1,
+        "connection_name": "balapvbyostoragecanary",
+    }
+
+    # Regular expression describing the pattern of an Application Insights connection string.
+    REGEX_APPINSIGHTS_CONNECTION_STRING = re.compile(
+        r"^InstrumentationKey=[0-9a-fA-F-]{36};IngestionEndpoint=https://.+.applicationinsights.azure.com/;LiveEndpoint=https://.+.monitor.azure.com/;ApplicationId=[0-9a-fA-F-]{36}$"
+    )
+
+    # Checks that a given dictionary has at least one non-empty (non-whitespace) string key-value pair.
+    @classmethod
+    def is_valid_dict(cls, d: dict[str, str]) -> bool:
+        return bool(d) and all(
+            isinstance(k, str) and isinstance(v, str) and k.strip() and v.strip() for k, v in d.items()
+        )
+
+    @classmethod
+    def validate_connection(
+        cls,
+        connection: Connection,
+        include_credentials: bool,
+        *,
+        expected_connection_type: Optional[ConnectionType] = None,
+        expected_connection_name: Optional[str] = None,
+        expected_authentication_type: Optional[CredentialType] = None,
+        expected_is_default: Optional[bool] = None,
+    ):
+        assert connection.id is not None
+
+        if expected_connection_name:
+            assert connection.name == expected_connection_name
+        else:
+            assert connection.name is not None
+
+        if expected_connection_type:
+            assert connection.type == expected_connection_type
+        else:
+            assert connection.type is not None
+
+        if expected_authentication_type:
+            assert connection.credentials.type == expected_authentication_type
+        else:
+            assert connection.credentials.type is not None
+
+        if expected_is_default is not None:
+            assert connection.is_default == expected_is_default
+
+        if include_credentials:
+            if type(connection.credentials) == ApiKeyCredentials:
+                assert connection.credentials.type == CredentialType.API_KEY
+                assert connection.credentials.api_key is not None
+
+    @classmethod
+    def validate_deployment(
+        cls,
+        deployment: Deployment,
+        *,
+        expected_model_name: Optional[str] = None,
+        expected_model_deployment_name: Optional[str] = None,
+        expected_model_publisher: Optional[str] = None,
+    ):
+        assert type(deployment) == ModelDeployment
+        assert deployment.type == DeploymentType.MODEL_DEPLOYMENT
+        assert deployment.model_version is not None
+        # Comment out the below, since I see that `Cohere-embed-v3-english` has an empty capabilities dict.
+        # assert TestBase.is_valid_dict(deployment.capabilities)
+        assert bool(deployment.sku)  # Check none-empty
+
+        if expected_model_name:
```

Comment:
nit: Should we create a method for these constructs?
```python
def _assety_equal_or_not_none(actual, expected=None)
    assert actual is not None
    if expected is not None:
        assert actual == expected
```

Reply:
Yes, good idea, I switched to using the function you suggested. Thanks!!